### PR TITLE
Ports fast process.

### PR DESCRIPTION
**Pull request by oisin100**
### Intent of your Pull Request

Ports Fastprocess and Super fun conveyor belts.
#### Changelog

:cl:
rscadd: Ported fast process from /TG/, Credit goes to @AndrewJacksonThe2nd
tweak: Conveyor belts no longer take forever to move stuff, infact they move items so quickly you can barely get a glimpse of them.
tweak: Ore redemption machines no longer take FOREVER to eat up ores.
/:cl:

I was tempted to commit my change for the singularity fast process... But that just turned the singularity into a Overpowered machine of pure destruction.

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
isn't that goofball

**Comment by ghost** [comment]
make sure that you port the foam fix too because this will fuck foam

**Comment by oisin100** [comment]
> make sure that you port the foam fix too because this will fuck foam

I did not port the FastProcess Foam(Thus it should not be borked), I like foam how it is.

**Comment by oisin100** [comment]
Heres a GIF of the conveyor belts, http://imgur.com/3DsnBq7

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -118,18 +118,25 @@ Class Procs:
...
+	var/fast_process = 0
 
 /obj/machinery/New()
 	..()
 	machines += src
-	SSmachine.processing += src
+	if(fast_process)
+		SSmachine.processing += src
+	else
+		SSfastprocess.processing += src
```
Comment (X-TheDark) [inline code/game/machinery/machinery.dm:129]: Other way around.

**Comment by X-TheDark** [comment]
I think this is too fast, honestly, maybe just set the wait to 10? (So belts are processed once every second instead of once every 2)

**Comment by ghost** [comment]
No, this is good

**Comment by X-TheDark** [comment]
While I don't mind making belts a bit faster, there must be SOME chance to ctrl+click someone or pick up an item without it whoosing by you at mach LUDICROUS.

File: code/game/machinery/machinery.dm (hunk 1 above)
Comment (oisin100) [inline code/game/machinery/machinery.dm:129]: Woops!

**Comment by oisin100** [comment]
> I think this is too fast, honestly, maybe just set the wait to 10? (So belts are processed once every second instead of once every 2)

Ill see what i can do to slow it down.

**Comment by MrsTonedOne** [comment]
Fair warning, @AndrewJacksonThe2nd is iamgoofball.

Also, resist any temptation to add more things to this. it can easily get clogged down and slow back down.

**Comment by X-TheDark** [comment]
Thanks for the warning, MSO.

Also, all you need to do to slow this down is to change `wait = 1` to `wait = 10` to slow it down to the extent I described.

**Comment by ghost** [comment]
NO.

I DO NOT WANT THIS SLOWED DOWN.

Conveyor belts are almost NEVER used for anything.

Just keep it at the current rate and merge it. 

@oisin100 

It's current speed is perfect and conveyor belts are never used so what's the harm?

**Comment by MrsTonedOne** [comment]
also, changing wait has no effect, this is a dynamic wait subsystem, so after every run, the mc dynamically reschedules it based on current load of the subsystem, the mc, and the game as a whole.

**Comment by Iamgoofball** [comment]
@oisin100 
@X-TheDark 

Hi, if you're trying to slow down fastprocess you don't understand how it works.

The thing about fastprocess is its literally trying to process as fast as possible WITHOUT causing lag.

Slowing down isn't possible without disabling the dynamic wait system, and is literally killing the point of it

PS. port the foam speedups too, the whole point of fastprocess was actually to speed up foam processing

**Comment by Iamgoofball** [comment]
> I was tempted to commit my change for the singularity fast process... But that just turned the singularity into a Overpowered machine of pure destruction.

we did some runs of this on Sybil, even with the lagfixes it gets excessive

**Comment by oisin100** [comment]
> Slowing down isn't possible without disabling the dynamic wait system, and is literally killing the point of it

You could use a cooldown(Not using Sleep god damn), and then mark objects as moved already so the next conveyor doesn't move the same object in the same iteration. I just haven't had the time to do it.

Anyways, closing this now, it will be done in the rebase.

**Pull request closed without merging**